Habitica/src/main/java/com/habitrpg/android/habitica/ui/viewHolders/tasks/BaseTaskViewHolder.java

```diff
@@ -87,14 +97,26 @@ public void bindHolder(Task newTask, int position) {
         if (this.rightBorderView != null) {
             this.rightBorderView.setBackgroundResource(this.task.getLightTaskColor());
         }
+	    if (task.getTags() != null && !task.tags.isEmpty())
+		    this.tagsGlyph.setVisibility(View.VISIBLE);
     }
 
+	@OnClick(R.id.tags_glyph)
+	void displayTagsToast(){
+		String tagString = "";
+		for(TaskTag t:
+		    task.getTags())
+		{
+			tagString = tagString + t.getTag().getName() + ", ";
+		}
+//		Toast.makeText(context, tagString, Toast.LENGTH_LONG).show();
```

Comment:
Ah, I see. Well, this issue probably needs some investigation then. The flow is the following:

1. Tag ids are returned on the task object from the API
2. Retrofit attempts to parse the tag and create a TaskTag model and that model attempts to link with the Task models: https://github.com/HabitRPG/habitica-android/blob/develop/Habitica/src/main/java/com/magicmicky/habitrpgwrapper/lib/models/tasks/TaskTag.java#L51

The Tag models are from Retrofit and the user response. So, maybe the Tags from the user are not being saved correctly. Or, there could be an issue with linking the TaskTag ides to local Tag models. Orrrr. This could be an issue with the deserializer : https://github.com/HabitRPG/habitica-android/tree/develop/Habitica/src/main/java/com/magicmicky/habitrpgwrapper/lib/utils. 

A lot going on there - sorry I don't have more insight.